### PR TITLE
fix: fix input, button and focus styles

### LINKED ISSUE #138: `InputLikeButton` display bugs
We have clashes with `button` in classless mode making the text white. See here https://codepen.io/team/tablecheck/pen/RwyYJye/ba7db9b07ddabac2ead9986a731938c6

Need to fix the alignment of `icon text`.

There's no way to toggle between "placeholder" and "value" styles.

CleanShot 2022-10-07 at 11 44 37

### PR DESCRIPTION
**Pull request by irmir**
Fixes #138 

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @tablecheck/tablekit-css@2.0.1-canary.146.3563899095.0
  npm install @tablecheck/tablekit-react-select@2.0.1-canary.146.3563899095.0
  npm install @tablecheck/tablekit-react@2.0.1-canary.146.3563899095.0
  # or 
  yarn add @tablecheck/tablekit-css@2.0.1-canary.146.3563899095.0
  yarn add @tablecheck/tablekit-react-select@2.0.1-canary.146.3563899095.0
  yarn add @tablecheck/tablekit-react@2.0.1-canary.146.3563899095.0
  ```
</details>

### CODE REVIEW

File: system/react/src/components/Input.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const baseStyles = css`
     box-shadow: 0 0 0 1px var(--border-active);
   }
 
-  &[data-error] {
+  &[data-error='true'] {
```
Comment (irmir) [inline system/react/src/components/Input.ts:41]: maybe we should apply this pattern to `data-disabled` below?

File: system/react/src/components/Button.ts [hunk 2]
```diff
@@ -17,9 +17,10 @@ export const classySelector = '.btn';
 export const baseStyles = css`
   position: relative;
   display: grid;
-  padding: 12px 20px;
+  padding: 14px var(--spacing-l4);
   grid-gap: var(--spacing-l2);
```
Comment (irmir) [inline system/react/src/components/Button.ts:21]: Screenshot 2022-11-21 at 19 14 20

File: system/react/src/components/Input.ts (hunk 1 above)
Comment (SashaShostyr) [reply system/react/src/components/Input.ts:41]: If we agree to use this here I think it should be changed for disabled state as well

Comment (SimeonC) [reply system/react/src/components/Input.ts:41]: I think for consistency we should probably change this to use a variant name. I don't like the `='true'` in CSS selectors/html as that becomes slightly unexpected in react.

```suggestion
  &[data-variant='error'] {
```

Comment (irmir) [reply system/react/src/components/Input.ts]: updated

File: system/react/src/components/Input.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export const baseStyles = css`
   font: var(--body-1);
   background: var(--surface);
   --width: 220px;
-  &[data-stretch] {
+  &[data-variant='stretch'] {
```
Comment (SimeonC) [inline system/react/src/components/Input.ts:15]: `data-stretch` is not the same as `data-variant` we can have `data-stretch data-variant="error"`.

`data-variant` should just be error and disabled.

**Review by SimeonC — CHANGES_REQUESTED** [review]

File: system/react/src/components/Input.ts (hunk 3 above)
Comment (irmir) [reply system/react/src/components/Input.ts]: fixed

File: package-lock.json [hunk 4]
```diff
@@ -12167,6 +12167,7 @@
       "version": "1.0.1",
       "resolved": "https://registry.npmjs.org/array.prototype.filter/-/array.prototype.filter-1.0.1.tgz",
       "integrity": "sha512-Dk3Ty7N42Odk7PjU/Ci3zT4pLj20YvuVnneG/58ICM6bt4Ij5kZaJTVQ9TSaWaIECX2sFyz4KItkVZqHNnciqw==",
+      "optional": true,
```
Comment (SashaShostyr) [inline package-lock.json:12170]: Why we have changes in this file?

**Review by SimeonC — APPROVED** [review]

File: package-lock.json (hunk 4 above)
Comment (irmir) [reply package-lock.json]: removed

**Review by SashaShostyr — DISMISSED** [review]

**Review by SimeonC — APPROVED** [review]

**Review by SimeonC — APPROVED** [review]
I think these changes fixes #138 as well?

**Comment by SashaShostyr** [comment]
> I think these changes fixes #138 as well?

Yes, should be fixed now

**Review by xsanctom — APPROVED** [review]

**Pull request merged**